server/resource_server/emails/sender/mailer_send_service.go

```diff
@@ -0,0 +1,71 @@
+package sender
+
+import (
+	"context"
+	"fmt"
+	"github.com/mailersend/mailersend-go"
+	"globe-and-citizen/layer8/server/resource_server/models"
+	"net/http"
+	"time"
+)
+
+const emailSendTimeout = time.Second * 10
+
+type MailerSendService struct {
+	apiKey     string
+	templateId string
+}
+
+func NewMailerSendService(apiKey string, templateId string) *MailerSendService {
+	ms := new(MailerSendService)
+	ms.apiKey = apiKey
+	ms.templateId = templateId
+	return ms
+}
+
+func (ms *MailerSendService) SendEmail(email *models.Email) error {
```

Comment:
can we pass context from the caller function instead of creating an entirely new context? its needed for tracing and passing data

Reply:
My goal was to make the email sender service to be agnostic to the type of email being sent, so it would be easier to reuse when we want to send other types of emails, e.g notifications.
I agree that for the current purposes we can pass the old context, but that would kind of limit the sender service functionality to sending verification codes specifically.

Reply:
its not related to logic, context is a variable that contains data from a request in golang, you can read that here https://medium.com/@jamal.kaksouri/the-complete-guide-to-context-in-golang-efficient-concurrency-management-43d722f6eaea

Reply:
sure lets just skip this for now